ext/php8/serializer.c

```diff
@@ -359,40 +359,40 @@ void ddtrace_serialize_span_to_array(ddtrace_span_fci *span_fci, zval *array) {
     add_assoc_long(el, "duration", span->duration);
 
     // SpanData::$name defaults to fully qualified called name (set at span close)
-    zval *prop_name = ddtrace_spandata_property_name(span->span_data);
+    zval *prop_name = ddtrace_spandata_property_name(span);
     zval prop_name_as_string;
-    if (Z_TYPE_P(prop_name) != IS_NULL) {
+    if (Z_TYPE_P(prop_name) > IS_NULL) {
         ddtrace_convert_to_string(&prop_name_as_string, prop_name);
         _add_assoc_zval_copy(el, "name", &prop_name_as_string);
     }
 
     // SpanData::$resource defaults to SpanData::$name
-    zval *prop_resource = ddtrace_spandata_property_resource(span->span_data);
-    if (Z_TYPE_P(prop_resource) != IS_NULL) {
+    zval *prop_resource = ddtrace_spandata_property_resource(span);
+    if (Z_TYPE_P(prop_resource) > IS_FALSE && (Z_TYPE_P(prop_resource) != IS_STRING || Z_STRLEN_P(prop_resource) > 0)) {
```

Comment:
This is technically a breaking change, but when I'm already touching this, it is in line with what userland does, unless we're going to prefer the current internal behaviour when moving userland spans to internal...